internal/resources/associate_role/resource.go

```diff
@@ -0,0 +1,245 @@
+		return utils.ProcessRemoteError(err)
+	})
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error creating associate role",
+			err.Error(),
+		)
+		return
+	}
+
+	current := NewAssociateRoleFromNative(associateRole)
+
+	diags = resp.State.Set(ctx, current)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Delete implements resource.Resource.
+func (r *associateRoleResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	// Get the current state.
+	var state AssociateRole
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	err := retry.RetryContext(
+		ctx,
+		5*time.Second,
+		func() *retry.RetryError {
+			_, err := r.client.AssociateRoles().
+				WithId(state.ID.ValueString()).
+				Delete().
+				Version(int(state.Version.ValueInt64())).
+				Execute(ctx)
+
+			return utils.ProcessRemoteError(err)
+		})
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error deleting associate role",
+			"Could not delete associate role, unexpected error: "+err.Error(),
+		)
+		return
+	}
+}
+
+// Metadata implements resource.Resource.
+func (*associateRoleResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_associate_role"
+}
+
+// Read implements resource.Resource.
+func (r *associateRoleResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	// Get the current state.
+	var state AssociateRole
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Read remote associate role and check for errors.
+	associateRole, err := r.client.AssociateRoles().WithId(state.ID.ValueString()).Get().Execute(ctx)
+	if err != nil {
+		if errors.Is(err, platform.ErrNotFound) {
+			resp.State.RemoveResource(ctx)
+			return
+		}
+
+		resp.Diagnostics.AddError(
+			"Error reading associate role",
+			"Could not retrieve the associate role, unexpected error: "+err.Error(),
+		)
+		return
+	}
+
+	// Transform the remote platform associate role to the
+	// tf schema matching representation.
+	current := NewAssociateRoleFromNative(associateRole)
+
+	// Set current data as state.
+	diags = resp.State.Set(ctx, &current)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Schema implements resource.Resource.
+func (*associateRoleResource) Schema(_ context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Associate Roles provide a way to group granular Permissions and assign " +
+			"them to Associates within a Business Unit.\n\n" +
+			"See also the [Associate Role API Documentation](https://docs.commercetools.com/api/projects/associate-roles)",
+		Version: 1,
```

Comment:
Can we remove this version field? Default terraform version is 0, so we can keep to that

Reply:
Preferably I would also have this method at the top, as it describes the actual schema the other methods use

Reply:
Understood, I was previously following the declaration order in the interface, but I'll certainly adhere to your request.